### PR TITLE
docs(adr): ADR-003: Remarkables contract

### PR DESCRIPTION
**Pull request by bragaz**
This PR contains the specification for the _Remarkables_ contract.

### CODE REVIEW

File: docs/architecture/adr-003-remarkables-contract.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+DRAFTED
+
+## Abstract
+This ADR defines the architecture of the Remarkables contract. This contract manage the creation of _Remarkables_.
+_Remarkables_ are Desmos Posts that has reached particular goals in terms of engagement (combination of reactions + comments).
+For example, a _remarkable_ post can be one that reached the goal of 50 comments and 100 reactions. The higher is the engagement
+goal reached, the rarer the _remarkable_ will be.
+
+## Context
+Applications like the new DFP are going to be a new definition of social apps merging classic social networks features to
```
Comment (dadamu) [inline docs/architecture/adr-003-remarkables-contract.md:17]: Can you write the full name of DFP here?

File: docs/architecture/adr-003-remarkables-contract.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  Config{},
+  EngagementThresholds{},
+  RarityMintFees{}
+}
+```
+
+#### Config
+The `Config{}` query returns the contract's configuration inside a `ConfigResponse`.
+```rust
+pub struct ConfigResponse {
```
Comment (dadamu) [inline docs/architecture/adr-003-remarkables-contract.md:138]:
```suggestion
pub struct QueryConfigResponse {
```

File: docs/architecture/adr-003-remarkables-contract.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  pub cw721_minter: Addr,
+  pub cw721_code_id: u64,
+  pub subspace_id: u64,
+}
+```
+
+#### EngagementThresholds
+The `EngagementThresholds` query returns the thresholds needed to mint a particular Remarkable inside a `EngagementThresholdsResponse`.
+```rust
+pub struct EngagementThresholdsResponse {
```
Comment (dadamu) [inline docs/architecture/adr-003-remarkables-contract.md:149]:
```suggestion
pub struct QueryEngagementThresholdsResponse {
```

File: docs/architecture/adr-003-remarkables-contract.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  pub engagement_thresholds: Vec<EngagementThreshold>
+}
+```
+
+#### RarityMintFees
+The `RarityMintFees` query return the rarity mint fees required to mint a specific Remarkable inside a `RarityMintFeesResponse`.
+```rust
+pub struct RarityMintFeesResponse {
```
Comment (dadamu) [inline docs/architecture/adr-003-remarkables-contract.md:157]:
```suggestion
pub struct QueryRarityMintFeesResponse {
```

**Review by dadamu — APPROVED** [review]
LGTM

**Review by manu0466 — APPROVED** [review]

**Pull request merged**